x-pack/plugins/monitoring/server/lib/create_query.js

```diff
@@ -89,9 +93,20 @@ export function createQuery(options) {
     combinedFilters.push(timeRangeFilter);
   }
 
-  return {
+  const query = {
     bool: {
       filter: combinedFilters.filter(Boolean)
     }
   };
+
+  if (isFromUnlinkedDeployment) {
+    query.bool.filter = [
+      ...query.bool.filter,
+      { bool: unlinkedDeploymentFilter }
```

Comment:
Given that `unlinkedDeploymentFilter` is `must: [ ... ]`, it could just be `[ ... ]` and then you could just do:

```diff
+...unlinkedDeploymentFilter
-{ bool: unlinkedDeploymentFilter }
```